flink-table/flink-table-planner/src/main/java/org/apache/flink/table/operations/TableAggFunctionCallFactory.java

```diff
@@ -0,0 +1,146 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.table.operations;
+
+import org.apache.flink.annotation.Internal;
+import org.apache.flink.api.common.typeinfo.TypeInformation;
+import org.apache.flink.api.common.typeinfo.Types;
+import org.apache.flink.table.api.ValidationException;
+import org.apache.flink.table.expressions.AggregateFunctionDefinition;
+import org.apache.flink.table.expressions.ApiExpressionDefaultVisitor;
+import org.apache.flink.table.expressions.ApiExpressionUtils;
+import org.apache.flink.table.expressions.CallExpression;
+import org.apache.flink.table.expressions.Expression;
+import org.apache.flink.table.expressions.ExpressionBridge;
+import org.apache.flink.table.expressions.FunctionDefinition;
+import org.apache.flink.table.expressions.PlannerExpression;
+import org.apache.flink.table.expressions.TableAggFunctionCall;
+import org.apache.flink.table.functions.TableAggregateFunction;
+
+import java.util.Collections;
+import java.util.List;
+
+import static java.util.stream.Collectors.toList;
+import static org.apache.flink.table.expressions.BuiltInFunctionDefinitions.AS;
+import static org.apache.flink.table.expressions.FunctionDefinition.Type.AGGREGATE_FUNCTION;
+
+/**
+ * Utility class for creating a valid {@link TableAggFunctionCall} operation.
+ */
+@Internal
+public class TableAggFunctionCallFactory {
```

Comment:
Why we need the factory class, can we add the `ExpressionBridge` as a parameter of `TableAggFunctionCallVisitor#createXX`  methods, or defined `bridge` as a member of `TableAggFunctionCallVisitor`?

Reply:
You are right. It's ok to remove the factory class and use visitor directly. I add this just to keep it consistent with the `CalculatedTableFactory` which is used to process udtf.